### ISSUE TITLE
Feature Request: Collaborative Tagging

### ISSUE THREAD

**Issue by Walkeras** [Microsoft/VoTT#752]
Hello,

It would be great if VoTT could be expanded to allow for Collaborative Tagging. There are multiple parts that I would imagine would be required:

- Identification of different taggers (user profiles)
- Comparing the tagging between taggers and tag validation
- General metrics on the tagging efficiency
- Quick updates to new/additional tags created
- Project permissions
- Communal tagging standard (share examples of what to tag within the project, might be a MD style document?)

**Comment by michalmar** [comment]
+1 especially for larger projects is needed

**Comment by devedse** [comment]
Implementing Collaborative tagging could also be done by simplifying the vott file.

For example, instead of using the security tokens and things like that, simple relative paths could be used. This would allow the users of this tool to upload the whole dataset to source control (e.g. git).

I think currently the tool tries to do more then is required. In my oppinion it should help do tagging and then allow for exporting in a specific format. Also copying the images again to the export directory is also now always the thing you want. For example if you want to have a simple CSV file that contains images + locations of the tags it would be way simpler to just generate that file with relative paths to the images.

**Comment by skywalkerisnull** [comment]
@devedse I don't think that we need a simplier vott file, what we would possibly need is a mapping function inside of vott that allows us to map the location of the data relative to the local machine/user. i.e. one computer might have a network drive mapped to "P" while another machine might have it mapped to "K".

There is also the ability to be doing this all through a web portal and having a login screen before hand that handles the security aspect of it, then it would access different blobs/storage containers hosted on Azure. This would be an ideal situation for a "hosted" solution.

When you say upload to git, do you mean all the raw data (i.e. images), or do you mean all the marks/tags that you have selected in the text based formats?



A simple way of doing this would allow us to do merging of projects, i.e. we divide the data to be tagged, we create a tag file that has all the different tags that we will use, and when we are done, we merge the project, at that point we could get the metrics. The hard part as an end user of VoTT in this would be ensuring that people get an even split based on their workloads, speed of processing/tagging, and how you would do near realtime QA/QC of the tagging. 

In addition, how you would manage conflicts with tags (same image, same object, different tag position), or how you would handle completly different tagging of in the same image. i.e. one image, one person tags 4 objects, the other person tags 6 objects etc.

**Comment by devedse** [comment]
@skywalkerisnull, I think having mappings and all that stuff just makes the whole process way more complex then it already is.

In my oppinion the VOTT tool is going past it's purpose of being a tagging tool and tries to do way more things then it should. Currently a lot of data is also duplicated (e.g. the id is in 3 places right now).

I guess I'll leave it up to the developers of the tool itself to make a decision but I think a short "re-evaluation" meeting might be in place.

To me something like this would make way more sense:

Project.vott:
```
...
"Images\Img1.png" {
   ....
}
...
```

Directory Structure:
```
Project.vott
Images (Folder)
    Img1.png
    Img2.png
```

This makes everything instantly portable for a file system.

If you want to focus more on colaboration as well I'd suggest not storing the images in the vott file at all, as you'd have multiple writes to the same file then at the same time.

In the .vott file only the root path of the images would be configured whereas the images itself could then be in a seperate directory with equally named .json files which contain tagging information. (They already do exist but are partial duplicates of the information in the .vott file).

Another solution could be is storing all the data in a cloud database but in my oppinion this would make the integration with any of the cloud platforms too tight.

Final note, I know this is not a final solution, but in my oppinion some re-evaluation might be needed to make the best out of this tool :)
